master/internal/resourcemanagers/job.go

```diff
@@ -0,0 +1,92 @@
+package resourcemanagers
+
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+)
+
+// FIXME haven't decided if resource manager actor should be responsible for this or not
+// we don't want a separate actor do we? could be useful for streaming job endpoints
+type GetJobOrder struct{}
+
+func filterAllocateRequests(reqs AllocReqs) AllocReqs {
+	isAdded := make(map[model.JobID]bool)
+	rv := make(AllocReqs, 0)
+	for _, req := range reqs {
+		job := req.Job
+		if job == nil {
+			continue
+		} else if _, ok := isAdded[job.JobID]; ok {
+			continue
+		}
+		isAdded[job.JobID] = true
+		rv = append(rv, req)
+	}
+	return rv
+}
+
+func allocReqsToJobSummaries(reqs AllocReqs) (summaries []*sproto.JobSummary) {
+	for _, req := range filterAllocateRequests(reqs) {
+		summaries = append(summaries, req.Job)
+	}
+	return summaries
+}
+
+// allocReqsToJobOrder convertes sorted allocation requests to job order.
+func allocReqsToJobOrder(reqs []*sproto.AllocateRequest) (jobIds []string) {
+	for _, req := range filterAllocateRequests(reqs) {
+		jobIds = append(jobIds, string(req.Job.JobID))
+	}
+	return jobIds
+}
+
+func allocateReqToV1Job(
+	rp *ResourcePool,
+	req *sproto.AllocateRequest,
+) (job *jobv1.Job) {
+	if req.Job == nil {
+		return job
+	}
+	group := rp.groups[req.Group]
+	job = &jobv1.Job{
+		Summary: &jobv1.JobSummary{
+			JobId: string(req.Job.JobID),
+			State: req.Job.State.Proto(), // look at AllocationState
+		},
+		EntityId:       req.Job.EntityID,
+		Type:           req.Job.JobType.Proto(),
+		IsPreemptible:  req.Preemptible,
+		ResourcePool:   req.ResourcePool,
+		User:           "hamid", // TODO
```

Comment:
nice

Reply:
haven't decided on a way to grab this yet. suggestions? I'm planning on putting it as out of scope for the PR so we can land and use the rest of the work.